herddb-collections/src/main/java/herddb/collections/TmpMap.java

```diff
@@ -0,0 +1,104 @@
+/*
+ Licensed to Diennea S.r.l. under one
+ or more contributor license agreements. See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership. Diennea S.r.l. licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing,
+ software distributed under the License is distributed on an
+ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied.  See the License for the
+ specific language governing permissions and limitations
+ under the License.
+
+ */
+package herddb.collections;
+
+/**
+ * This Map is useful to store an huge amount of data locally, like caching the results of an huge record set. We don't
+ * want to fill the heap with data. The underlying HerdDB database will swap to disk data that are not needed.
+ * <p>
+ * Null keys and null values are not supported.
+ * <p>
+ * The expected workflow is:
+ * <ul>
+ * <li>Load the map
+ * <li>Perform lookups
+ * <li>Close the map
+ * </ul>.
+ * <p>
+ * <b>The map is not expected to be thread safe. Do not wrap it with an external synchronization mechanism as it is
+ * anyway sharing internal data structures with every other collection allocated by the same CollectionsManager</b>
+ *
+ * @author enrico.olivelli
+ */
+public interface TmpMap<K, V> extends AutoCloseable {
```

Comment:
Should there be support for things like computeIfAbsent, putifAbsent

Reply:
Good idea
We can do it in the future.
If you think about a non thread safe implementation you can implement it very easily.
But if we go thru a thread safe implementation we need to be careful for computeXxxx.

putIfAbsent is easy, it is like current put() but without the UPDATE.

Actually I don't need such features.
I prefer to add features only if needed: once you release a public API you must maintain it for long term